src/test/scala/io/confluent/examples/streams/ProbabilisticCountingScalaIntegrationTest.scala

```diff
@@ -99,63 +84,52 @@ class ProbabilisticCountingScalaIntegrationTest extends AssertionsForJUnit {
     builder.addStateStore(createCMSStoreBuilder(cmsStoreName))
 
     // Read the input from Kafka.
-    val textLines: KStream[Array[Byte], String] = builder.stream(inputTopic)
+    val textLines: KStream[String, String] = builder.stream(inputTopic)
 
     // Scala-Java interoperability: to convert `scala.collection.Iterable` to  `java.util.Iterable`
     // in `flatMapValues()` below.
     import collection.JavaConverters.asJavaIterableConverter
 
     val approximateWordCounts: KStream[String, Long] = textLines
-        .flatMapValues(textLine => textLine.toLowerCase.split("\\W+").toIterable.asJava)
-        .transform(
-          new TransformerSupplier[Array[Byte], String, KeyValue[String, Long]] {
-            override def get() = new ProbabilisticCounter(cmsStoreName)
-          },
-          cmsStoreName)
+      .flatMapValues(textLine => textLine.toLowerCase.split("\\W+").toIterable.asJava)
+      .transform(
+        new TransformerSupplier[String, String, KeyValue[String, Long]] {
+          override def get() = new ProbabilisticCounter(cmsStoreName)
+        },
+        cmsStoreName)
 
     // Trick to re-use Kafka's serde for java.lang.Long for scala.Long.
     val longSerde: Serde[Long] = Serdes.Long().asInstanceOf[Serde[Long]]
```

Comment:
Unrelated note: We should no longer need this trick.  With KStreams native Scala API, we don't need to set serdes explicitly anymore.

Reply:
Same explanation as above. This will be true once we port to the scala dsl.

Reply:
@miguno , during the downstream merge, I realize this was added in a later release, so your feedback is already implemented in downstream branches.